WordPress/src/main/java/org/wordpress/android/ui/sitecreation/NewSiteCreationMainVM.kt

```diff
@@ -82,10 +92,27 @@ class NewSiteCreationMainVM @Inject constructor(
         wizardManager.showNextStep()
     }
 
-    fun shouldSuppressBackPress(): Boolean = wizardManager.isLastStep()
-
-    fun onBackPressed() {
-        wizardManager.onBackPressed()
+    /**
+     * Returns true if the back pressed event was handled and the activity should suppress it, false otherwise.
+     */
+    fun onBackPressed(): Boolean {
```

Comment:
It's most likely just me, but I had a hard time wrapping my head around understanding the code without reading the comment. I know this is a convention in Java, but I think it's mostly because the method is both handling something and then also returning whether the caller should also handle it. Just thinking out loud here, I am curious if it would be better to not return anything from this method and trigger an event that'll be observed by the activity to actually go back.

Reply:
Tbh I'm not a big fan of this pattern either. I used it just because it's a pretty common pattern on Android. However, I agree with you. Updated in #[d987743](https://github.com/wordpress-mobile/WordPress-Android/pull/9187/commits/d9877432b1f3788866aba3815b04f0bb1cf33f9f).